### PR TITLE
Drop Qt5 support

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: src/app/CMakeLists.txt [hunk 1]
```diff
@@ -14,16 +14,12 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
 endif()
 
 # lrelease .ts files
-find_package(${QT_PKG} COMPONENTS LinguistTools)
+find_package(Qt COMPONENTS LinguistTools)
```
Comment (m-kuhn) [inline src/app/CMakeLists.txt:17]: Note to self: do versionless `find_package` calls work? To be checked

File: src/app/CMakeLists.txt [hunk 2]
```diff
@@ -113,24 +103,20 @@ function(create_executable)
...
-               ${QT_PKG}::QDarwinMicrophonePermissionPlugin
-               ${QT_PKG}::QDarwinLocationPermissionPlugin)
-      qt_import_plugins(
-        ${exe_TARGET} INCLUDE ${QT_PKG}::QDarwinCameraPermissionPlugin
-        ${QT_PKG}::QDarwinMicrophonePermissionPlugin
-        ${QT_PKG}::QDarwinLocationPermissionPlugin)
-    endif()
-  else()
-    qt5_import_plugins(${exe_TARGET} INCLUDE ${QT_PKG}::QSvgPlugin)
+  target_link_libraries(${exe_TARGET} PUBLIC Qt6::Core5Compat)
```
Comment (m-kuhn) [inline src/app/CMakeLists.txt:106]:
```suggestion
  target_link_libraries(${exe_TARGET} PUBLIC Qt::Core5Compat)
```

File: src/qml/CMakeLists.txt [hunk 3]
```diff
@@ -1,31 +1,20 @@
...
-      QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qml.qrc
-      ${CMAKE_CURRENT_SOURCE_DIR}/qmlqt6.qrc)
-  else()
-    qtquick_compiler_add_resources(
-      QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qml.qrc
-      ${CMAKE_CURRENT_SOURCE_DIR}/qmlqt5.qrc)
-  endif()
+  qtquick_compiler_add_resources(
+    QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qml.qrc
+    ${CMAKE_CURRENT_SOURCE_DIR}/qmlqt6.qrc)
```
Comment (m-kuhn) [inline src/qml/CMakeLists.txt:10]:
```suggestion
    )
```

File: src/qml/CMakeLists.txt [hunk 4]
```diff
@@ -1,31 +1,20 @@
...
 else()
-  if(BUILD_WITH_QT6)
-    qt_add_resources(QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qml.qrc)
-    qt_add_resources(QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qmlqt6.qrc)
-  else()
-    qt5_add_resources(QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qml.qrc)
-    qt5_add_resources(QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qmlqt5.qrc)
-  endif()
+  qt_add_resources(QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qml.qrc)
+  qt_add_resources(QML_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/qmlqt6.qrc)
```
Comment (m-kuhn) [inline src/qml/CMakeLists.txt:13]:
```suggestion
```

File: cmake/qgis-cmake-wrapper.cmake [hunk 5]
```diff
@@ -136,42 +136,28 @@ if(TRUE) # Should possibly have a "static only" check
...
+      Qt::Sql
+      Qt::Positioning
+      Qt::Core5Compat
     )
-  if(BUILD_WITH_QT6)
-    find_package(${QT_PKG} COMPONENTS Core5Compat)
-    target_link_libraries(QGIS::Core INTERFACE
-      ${QT_PKG}::Core5Compat
-      )
   endif()
```
Comment (m-kuhn) [inline cmake/qgis-cmake-wrapper.cmake:153]:
```suggestion
```

File: cmake/qgis-cmake-wrapper.cmake [hunk 6]
```diff
@@ -136,42 +136,27 @@ if(TRUE) # Should possibly have a "static only" check
...
   target_link_libraries(QGIS::Core INTERFACE PkgConfig::freexl)
-  if(BUILD_WITH_QT6)
-    _qgis_core_add_dependency(Qt6Keychain::Qt6Keychain Qt6Keychain)
-  else()
-    _qgis_core_add_dependency(Qt5Keychain::Qt5Keychain Qt5Keychain)
-  endif()
+  _qgis_core_add_dependency(Qt6Keychain::Qt6Keychain Qt6Keychain)
 
-  find_package(${QT_PKG} COMPONENTS Core Gui Network Xml Svg Concurrent Sql Positioning)
+  find_package(Qt COMPONENTS Core Gui Network Xml Svg Concurrent Sql Positioning Core5Compat)
```
Comment (m-kuhn) [inline cmake/qgis-cmake-wrapper.cmake:141]:
```suggestion
  find_package(Qt6 COMPONENTS Core Gui Network Xml Svg Concurrent Sql Positioning Core5Compat)
```

File: cmake/qgis-cmake-wrapper.cmake [hunk 7]
```diff
@@ -136,42 +136,27 @@ if(TRUE) # Should possibly have a "static only" check
...
     )
-  if(BUILD_WITH_QT6)
-    find_package(${QT_PKG} COMPONENTS Core5Compat)
-    target_link_libraries(QGIS::Core INTERFACE
-      ${QT_PKG}::Core5Compat
-      )
-  endif()
   if(NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
-    find_package(${QT_PKG} COMPONENTS SerialPort)
+    find_package(Qt COMPONENTS SerialPort)
```
Comment (m-kuhn) [inline cmake/qgis-cmake-wrapper.cmake:154]:
```suggestion
    find_package(Qt6 COMPONENTS SerialPort)
```

File: cmake/qgis-cmake-wrapper.cmake [hunk 8]
```diff
@@ -183,9 +168,9 @@ if(TRUE) # Should possibly have a "static only" check
      # _find_and_link_library(lcms2 QGIS::Core)
 
     # QtKeychain
-    find_package(${QT_PKG} COMPONENTS DBus REQUIRED)
+    find_package(Qt COMPONENTS DBus REQUIRED)
```
Comment (m-kuhn) [inline cmake/qgis-cmake-wrapper.cmake:171]:
```suggestion
    find_package(Qt6 COMPONENTS DBus REQUIRED)
```

File: src/app/CMakeLists.txt [hunk 9]
```diff
@@ -1,4 +1,4 @@
-find_package(${QT_PKG}QuickCompiler QUIET)
+find_package(QtQuickCompiler QUIET)
```
Comment (m-kuhn) [inline src/app/CMakeLists.txt:1]:
```suggestion
find_package(Qt6QuickCompiler QUIET)
```

File: src/app/CMakeLists.txt (hunk 1 above)
Comment (m-kuhn) [inline src/app/CMakeLists.txt:17]:
```suggestion
find_package(Qt6 COMPONENTS LinguistTools)
```

File: src/app/CMakeLists.txt [hunk 10]
```diff
@@ -89,12 +79,12 @@ function(create_executable)
...
   endif()
 
-  get_target_property(QT_TARGET_TYPE ${QT_PKG}::Core TYPE)
+  get_target_property(QT_TARGET_TYPE Qt::Core TYPE)
   # This should be in src/qml/CMakeLists.txt but that's not possible with Qt5.
   # Retry with Qt6 Same is in tests/CMakeLists.txt
   # https://bugreports.qt.io/browse/QTBUG-80847
   if(${QT_TARGET_TYPE} STREQUAL "STATIC_LIBRARY")
-    find_package(${QT_PKG} COMPONENTS QmlImportScanner)
+    find_package(Qt COMPONENTS QmlImportScanner)
```
Comment (m-kuhn) [inline src/app/CMakeLists.txt:87]:
```suggestion
    find_package(Qt6 COMPONENTS QmlImportScanner)
```

File: src/core/CMakeLists.txt [hunk 11]
```diff
@@ -344,40 +337,34 @@ target_link_libraries(
 
 if(WITH_BLUETOOTH)
   find_package(
-    ${QT_PKG}
+    Qt
```
Comment (m-kuhn) [inline src/core/CMakeLists.txt:340]:
```suggestion
    Qt6
```

File: src/core/CMakeLists.txt [hunk 12]
```diff
@@ -344,40 +337,34 @@ target_link_libraries(
...
     COMPONENTS Bluetooth
     REQUIRED)
-  target_link_libraries(qfield_core PUBLIC ${QT_PKG}::Bluetooth)
+  target_link_libraries(qfield_core PUBLIC Qt::Bluetooth)
 endif()
 
 if(WITH_SERIALPORT)
   find_package(
-    ${QT_PKG}
+    Qt
```
Comment (m-kuhn) [inline src/core/CMakeLists.txt:348]:
```suggestion
    Qt6
```

File: src/core/CMakeLists.txt [hunk 13]
```diff
@@ -344,40 +337,34 @@ target_link_libraries(
...
     COMPONENTS SerialPort
     REQUIRED)
-  target_link_libraries(qfield_core PUBLIC ${QT_PKG}::SerialPort)
+  target_link_libraries(qfield_core PUBLIC Qt::SerialPort)
 endif()
 
 if(WITH_NFC)
   find_package(
-    ${QT_PKG}
+    Qt
```
Comment (m-kuhn) [inline src/core/CMakeLists.txt:356]:
```suggestion
    Qt6
```

File: src/qml/CMakeLists.txt [hunk 14]
```diff
@@ -1,31 +1,19 @@
-find_package(${QT_PKG}QuickCompiler QUIET)
-find_package(${QT_PKG} COMPONENTS Quick)
+find_package(QtQuickCompiler QUIET)
+find_package(Qt COMPONENTS Quick)
```
Comment (m-kuhn) [inline src/qml/CMakeLists.txt:2]:
```suggestion
find_package(Qt6QuickCompiler QUIET)
find_package(Qt6 COMPONENTS Quick)
```

File: test/CMakeLists.txt [hunk 15]
```diff
@@ -45,9 +45,9 @@ macro (ADD_QFIELD_QML_TEST TESTNAME TESTSRC)
   # This should be in src/qml/CMakeLists.txt but that's not possible with Qt5. Retry with Qt6
   # Same is in src/app/CMakeLists.txt
   # https://bugreports.qt.io/browse/QTBUG-80847
-  get_target_property(QT_TARGET_TYPE ${QT_PKG}::Core TYPE)
+  get_target_property(QT_TARGET_TYPE Qt::Core TYPE)
   if(${QT_TARGET_TYPE} STREQUAL "STATIC_LIBRARY")
-    find_package(${QT_PKG} COMPONENTS QmlImportScanner)
+    find_package(Qt COMPONENTS QmlImportScanner)
```
Comment (m-kuhn) [inline test/CMakeLists.txt:50]:
```suggestion
    find_package(Qt6 COMPONENTS QmlImportScanner)
```

File: src/core/CMakeLists.txt [hunk 16]
```diff
@@ -344,40 +334,34 @@ target_link_libraries(
 
 if(WITH_BLUETOOTH)
   find_package(
-    ${QT_PKG}
+    Qt6
     COMPONENTS Bluetooth
     REQUIRED)
-  target_link_libraries(qfield_core PUBLIC ${QT_PKG}::Bluetooth)
+  target_link_libraries(qfield_core PUBLIC Qt::Bluetooth)
```
Comment (m-kuhn) [inline src/core/CMakeLists.txt:340]:
```suggestion
  target_link_libraries(qfield_core PUBLIC Qt6::Bluetooth)
```

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1c543f36cf8c6b11d5c48eb43fce1b9d4b6edfd9: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5172-064526c93152db719dabd3efd27e30f2c6efa269-android-x86.apk)
</details>

**Review by nirvn — APPROVED** [review]
Android APK working like a charm.

Bye bye Qt5.

**Pull request merged**